ElrondNetwork/elrond-go
# EN-8910: encapsulated the API services into wrapper

Created an API wrapper so we can change the facade implementation while the node is running. Useful for example when the epoch bootstrapping is in progress and the API endpoints should return a message signaling that the node is starting, rather than API service not being started at all.

Testing procedure:
1. start a testnet
2. check that the API is working properly. Check `api-interface/network/config` for example.
3. when a node is shuffled out, check it's API (can be the same endpoint from point 2.). It should return mocked messages on all routes telling that the node is starting. 
3.1. Alternatively, you can start a new node and it should sync using the `start in epoch` feature. During the sync, the API should return mocked messages.
4. after the sync after shuffle out is done, the API should return correct data again
